wasm/src/graphics/webgl.rs

```diff
@@ -0,0 +1,126 @@
+use super::webgl_bindings::Gl2;
+use super::GraphicsApi;
+use crate::error::ClientError;
+use rask_engine::{math::Mat3, resources::Texture};
+
+mod imports {
+    extern "C" {
+        pub fn get_canvas_size() -> u32;
+        pub fn set_canvas_size(w: u32, h: u32);
+    }
+}
+
+fn get_canvas_size() -> (u32, u32) {
+    let x = unsafe { imports::get_canvas_size() };
+    (x >> 16, x & 0xffff)
+}
+
+fn set_canvas_size(w: u32, h: u32) {
+    unsafe { imports::set_canvas_size(w, h) }
+}
+
+#[derive(Debug)]
+pub enum WebGl2Error {
+    ContextLost,
+    InvalidEnum,
+    InvalidValue,
+    InvalidOperation,
+    InvalidFramebufferOperation,
+    OutOfMemory,
+    UnknownError,
```

Comment:
should be 
`ContextLost = Gl2::CONTEXT_LOST,`

Reply:
What would be the advantage? You can not cast an integer to an enum.

Reply:
to ensure consistency in the codebase. Were you to change the constants or someone without experience in webgl changes the order of the enum, that could lead so some pretty nasty to debug errors. It does not cost anything, is less prone to errors and is less confusing, because you are not defining the error codes in two parts of the codebase.
Someone might expect that the enum variants represent the error codes which is not guaranteed